src/main/scala/io/picnicml/doddlemodel/preprocessing/Binarizer.scala

```diff
@@ -0,0 +1,40 @@
+package io.picnicml.doddlemodel.preprocessing
+
+import breeze.linalg.{*, DenseVector}
+import io.picnicml.doddlemodel.data.Feature.FeatureIndex
+import io.picnicml.doddlemodel.data.{Features, RealVector}
+import io.picnicml.doddlemodel.typeclasses.Transformer
+
+case class Binarizer private (private val thresholds: RealVector, private val featureIndex: FeatureIndex) {
+  private val numNumeric = featureIndex.numerical.columnIndices.length
+  require(numNumeric == 0 || numNumeric == thresholds.length, "A threshold should be given for every numerical column")
+}
+
+object Binarizer {
+
+  def apply(threshold: Double, featureIndex: FeatureIndex): Binarizer = {
+    val numNumeric: Int = featureIndex.numerical.columnIndices.length
+    val thresholdsExtended = DenseVector.fill(numNumeric) {threshold}
+    Binarizer(thresholdsExtended, featureIndex)
+  }
+
+  implicit lazy val ev: Transformer[Binarizer] = new Transformer[Binarizer] {
+
+    override def isFitted(model: Binarizer): Boolean = true
+
+    override def fit(model: Binarizer, x: Features): Binarizer = model
+
+    override protected def transformSafe(model: Binarizer, x: Features): Features = {
```

Comment:
Ah, I should have known there was a good reason the code in other preprocessors looked more complex :sweat_smile: 

You are right, I will fix it like you suggested. One question though, is the `zipWithIndex` really necessary? As far as I can tell, `colIndex` and `thresholdIndex` will always be the same.

Reply:
Columns referenced in `featureIndex` and columns in the actual data matrix might not be the same. You can think of `featureIndex` as a way of saying "we are talking about these very specific columns" (e.g. data matrix `x` has 4 columns and `featureIndex.columnIndices` is `IndexedSeq(0, 2)` meaning we are only talking about the first and the third column). Namely, an instance of `featureIndex` is passed to preprocessors and enables the ability to preprocess only a subset of columns. This should be implemented in a more explicit manner in the future, I know :).

As a result `colIndex` and `thresholdIndex` shouldn't alway be the same above. Even if an instance of `featureIndex` that specifies all columns is passed to the preprocessor, the categorical columns are ignored by `.numerical` which will result in `colIndex` and `thresholdIndex` being out of sync, because `thresholds` only contains thresholds for numerical columns. Hope that makes sense, LMK if I'm being unclear.